### PR TITLE
Settling the auction

### LINKED ISSUE #1742: Auction Winner Settling the Auction
## Context

This issue encapsulates design and development of settling the auction by the winner. 

After auction is won, the user has to _actively_ settle the auction for the ownership transfer to be recorded on the chain. This transaction of ownership transfer has a fee applying to it, paid by the winner.  In fact this can be done by any agent, but we are restricting it to the winner in the UI. 

🎥  [Video walkthrough](https://www.loom.com/share/4cc368aad7ab4f498c2271a6831d8ac3)

## AC
- [ ] Notification event - auction won - snack bar deep-linking to auction settlement page
- [ ] Auction settled confirmation snackbar
- [ ] Handling the transaction
- [ ] Animation Confetti on the winner's view

## Designs

🎨 [ LInk to Figma ](https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-Purchase-view?node-id=1237%3A23354)

### PR DESCRIPTION
**Pull request by drillprop**
Fix #1742 
Designs: https://www.figma.com/file/wjeAOZq77pSUFZsYXJqKIB/NFT-Purchase-view?node-id=1237%3A23354

### CODE REVIEW

File: packages/atlas/src/views/viewer/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { useAsset, useMemberAvatar } from '@/providers/assets'
+import { useJoystream } from '@/providers/joystream'
+import { useSnackbar } from '@/providers/snackbars'
+import { useTransaction } from '@/providers/transactionManager'
+import { useAuthorizedUser } from '@/providers/user'
+
+import { Content, StyledLayoutGrid, StyledLimitedContainer, StyledLottie } from './NftSettlementBottomDrawer.styles'
+
+type NftSettlementBottomDrawerProps = {
+  isOpen: boolean
```
Comment (kdembler) [inline packages/atlas/src/views/viewer/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx]: Could you hook this up with the `NftActionsContext`?

Comment (drillprop) [reply packages/atlas/src/views/viewer/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx]: done!

**Comment by kdembler** [comment]
The app doesn't load for me, getting an error:
`Error: Trying to use authorized user without authorization`

File: packages/atlas/src/App.tsx [hunk 2]
```diff
@@ -10,6 +10,7 @@ import { NftActionsProvider } from '@/providers/nftActions'
 import { GlobalStyles } from '@/styles'
 import { NftSaleBottomDrawer } from '@/views/global/NftSaleBottomDrawer'
 import { NftPurchaseView } from '@/views/viewer/NftPurchaseView'
+import { NftSettlementBottomDrawer } from '@/views/viewer/NftSettlementBottomDrawer'
```
Comment (kdembler) [inline packages/atlas/src/App.tsx]: Can we move it to `@views/global/` as well?

File: packages/atlas/src/components/NftCard/Members.tsx [hunk 3]
```diff
@@ -12,22 +12,25 @@ import {
   StyledAvatarGroup,
 } from './NftCard.styles'
 
+import { SkeletonLoader } from '../_loaders/SkeletonLoader'
```
Comment (kdembler) [inline packages/atlas/src/components/NftCard/Members.tsx]: Should probably be absolute import

File: packages/atlas/src/components/NftCard/NftCard.tsx [hunk 4]
```diff
@@ -5,30 +5,45 @@ import { VideoThumbnail, VideoThumbnailProps } from '@/components/_video/VideoTh
 import { Member, Members } from './Members'
 import { Container, Content, Details, Separator, Title } from './NftCard.styles'
 
+import { SkeletonLoader } from '../_loaders/SkeletonLoader'
```
Comment (kdembler) [inline packages/atlas/src/components/NftCard/NftCard.tsx]: Same here

File: packages/atlas/src/components/NftCard/NftCard.tsx [hunk 5]
```diff
@@ -5,30 +5,45 @@ import { VideoThumbnail, VideoThumbnailProps } from '@/components/_video/VideoTh
...
+  fullWidth,
+  loading,
+}) => {
   return (
     <Container fullWidth={fullWidth}>
       <VideoThumbnail clickable={false} thumbnailUrl={thumbnail.thumbnailUrl} />
       <Details>
-        <Title variant="h400">{title}</Title>
+        {loading ? (
+          <SkeletonLoader width="55.6%" height={24} bottomSpace={24} />
```
Comment (kdembler) [inline packages/atlas/src/components/NftCard/NftCard.tsx]: Where is this percentage coming from?

Comment (drillprop) [reply packages/atlas/src/components/NftCard/NftCard.tsx]: To be honest, don't know - I just copied this value from `NftTile`. I'll set this to the rounded value.

**Comment by drillprop** [comment]
> The app doesn't load for me, getting an error: `Error: Trying to use authorized user without authorization`

I used the wrong hook, it should work now

File: packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            <Content>
+              <Text variant="h600">You have won the auction! 🎉</Text>
+              <Text variant="t300" secondary margin={{ top: 4, bottom: 10 }}>
+                Congratulations! To update the ownership, you need to settle the auction.
+              </Text>
+              <Button size="large" fullWidth={!xsMatch} onClick={handleSettleAuction}>
+                Settle the auction
+              </Button>
+              <Text variant="t100" secondary margin={{ top: 4 }}>
+                Transaction fee: <Text variant="t100">19 tJoy</Text>
```
Comment (kdembler) [inline packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx]:
```suggestion
                Transaction fee: <Text variant="t100">0 tJOY</Text>
```

File: packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      onTxSync: async (_) => {
+        closeNftAction()
+      },
+      onTxFinalize: () => {
+        displaySnackbar({
+          title: 'Auction settled',
+          description: 'Your auction has been settled. You are now the owner of this NFT',
+          iconType: 'success',
+        })
+        return refetch()
```
Comment (kdembler) [inline packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx:53]: I think we also should close the drawer once the sync has finished?

**Review by kdembler — APPROVED** [review]
Two small comments, otherwise works sweet! 🥇

File: packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+            <Content>
+              <Text variant="h600">You have won the auction! 🎉</Text>
+              <Text variant="t300" secondary margin={{ top: 4, bottom: 10 }}>
+                Congratulations! To update the ownership, you need to settle the auction.
+              </Text>
+              <Button size="large" fullWidth={!xsMatch} onClick={handleSettleAuction}>
+                Settle the auction
+              </Button>
+              <Text variant="t100" secondary margin={{ top: 4 }}>
+                Transaction fee: <Text variant="t100">0 tJoy</Text>
```
Comment (kdembler) [inline packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx]: Let's try this again :D
```suggestion
                Transaction fee: <Text variant="t100">0 tJOY</Text>
```

File: packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+        <StyledLayoutGrid>
+          <GridItem rowStart={{ base: 2, sm: 1 }} colSpan={{ base: 12, sm: 6, md: 5, lg: 4 }} colStart={{ lg: 3 }}>
+            <NftCard
+              title={nft?.video.title}
+              thumbnail={{
+                loading: thumbnailLoading,
+                thumbnailUrl: thumbnailUrl,
+              }}
+              creator={{ name: nft?.video.channel.title, assetUrl: avatarUrl }}
+              owner={{ name: lastBidder?.handle, assetUrl: memberAvatarUrl }}
```
Comment (kdembler) [inline packages/atlas/src/views/global/NftSettlementBottomDrawer/NftSettlementBottomDrawer.tsx]: Let's use the current owner instead

**Pull request merged**